hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -152,7 +153,8 @@ protected Option<HoodieRecord<HoodieMetadataPayload>> getRecordByKey(String key,
   @Override
   public List<String> getPartitionPathsWithPrefixes(List<String> prefixes) throws IOException {
     return getAllPartitionPaths().stream()
-        .filter(p -> prefixes.stream().anyMatch(p::startsWith))
+        .filter(p -> prefixes.stream().anyMatch(queryPaths ->
```

Comment:
@onlywangyh this is not the right place to fix this -- this should be fixed on the caller side: if caller specifies prefix w/o a slash this method will match it appropriately. Instead, caller should have specified a path ending with a slash to signal that it doesn't want any other folders to be matched